src/main/java/net/minecraftforge/fml/common/registry/VillagerRegistry.java

```diff
@@ -306,10 +392,54 @@ public boolean equals(Object o)
      */
     public static void setRandomProfession(EntityVillager entity, Random rand)
     {
-        Set<String> entries = INSTANCE.professions.getKeys();
+        ArrayList<String> entries = new ArrayList(INSTANCE.professionsByName.keySet());
         int prof = rand.nextInt(entries.size());
-        //TODO: Grab id range from internal registry
-        entity.setProfession(rand.nextInt(5));
+        VillagerProfession profession = INSTANCE.professionsByName.get(entries.get(prof));
+        entity.setProfession(profession.id);
+    }
+    
+    public static void populateBuyingList(MerchantRecipeList recipeList, int professionID, int careerId, int careerLevel, Random random)
+    {
+        VillagerProfession profession = INSTANCE.professionsById.get(professionID);
+        //The villagers seem to have career = 0 meaning no career.
+        if(careerId > profession.careers.size())
```

Comment:
that method is always called after the villager increments their career to 1 (or higher).  the way I wrote it, if the profession.careers is empty, then it will return with no trades added, as the value for careerId passed in is always > 0.